go.mod

```diff
@@ -22,6 +22,7 @@ require (
 	github.com/hashicorp/go-version v1.7.0
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.9.0
+	github.com/xwb1989/sqlparser v0.0.0-20180606152119-120387863bf2
```

Comment:
Could we instead use the code this was based on: https://github.com/vitessio/vitess/tree/main/go/vt/sqlparser

It seems like this was a fork of that repository[^1], but it hasn't been maintained or synced since that fork.

Would it be too large of a dependency to just rely on vitess directly?

[^1]: https://github.com/xwb1989/sqlparser?tab=readme-ov-file#notice

Reply:
I did think about using vitess directly, but it seemed like a lot more than what we needed. Since this is just an implementation detail, we could refactor it later if we want but I think this is the best option for right now

Reply:
I'm concerned when looking through all the changes that have been applied to the upstream code over the past 6 years, there's a considerable amount of changes. Notably:

- There are many additional statement types that can be parsed: https://pkg.go.dev/vitess.io/vitess/go/vt/sqlparser#StatementType
- The statement type itself can print its name meaning we will not need to add an additional switch statement: https://pkg.go.dev/vitess.io/vitess/go/vt/sqlparser#StatementType.String
- The code is not owned by Google, it is licensed by a CNCF project
- https://github.com/vitessio/vitess/issues/4174
- https://github.com/vitessio/vitess/pull/2395 which addressed:
  - https://github.com/vitessio/vitess/issues/1953
  - https://github.com/vitessio/vitess/issues/2184
- https://github.com/vitessio/vitess/pull/8307

Should we make our own fork of this pacakge?

Reply:
Can we maybe get a better understanding of dependency size by looking at what the binary size is based on the possible dependencies?

Reply:
I'll admit it looks like you've dug into it deeper than I did, hah. These are some great points so maybe that is the better approach. I'll try switching it to that

Reply:
The binary size went from 30MB->37MB with vitess included, that's not small but I'm not personally concerned about it and it could be something we look to refactor later. wdyt?